src/GraphQL/TypeApi.hs

```diff
@@ -50,17 +50,23 @@ queryError = throwM . QueryError
 -- NormalizedSelectionSet which has query fragments etc. resolved.
 class (MonadThrow m, MonadIO m) => HasGraph m a where
   type HandlerType m a
-  buildResolver :: HandlerType m a -> AST.SelectionSet -> m GValue.Value
+  buildResolver :: HandlerType m a -> CanonicalQuery -> m GValue.Value
 
--- Parse a value of the right type from an argument
--- TODO
+-- | The ReadValue instance converts AST.Value types like ValueInt to
+-- the type expected by the handler function. It's the boundary
+-- between incoming types and your custom application Haskell types.
 class ReadValue a where
```

Comment:
+1 on naming. I we have a slightly different API with an extra "missingValue" function which would be useful for default values. If you have bandwidth please change naming as you see fit.